lib/beautiful/gtk.lua

```diff
@@ -0,0 +1,174 @@
+---------------------------------------------------------------------------
+--- Querying current GTK+ 3 theme via GtkStyleContext.
+--
+-- @author Yauheni Kirylau &lt;yawghen@gmail.com&gt;
+-- @copyright 2016-2017 Yauheni Kirylau
+-- @module beautiful.gtk
+---------------------------------------------------------------------------
+local gears_debug = require("gears.debug")
+local gears_math = require("gears.math")
+local join = require("gears.table").join
+local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
+
+
+local gtk = {
+  cached_theme_variables = nil
+}
+
+
+local function convert_gtk_channel_to_hex(channel_value)
+  return string.format("%02x", gears_math.round(channel_value * 255))
+end
+
+local function convert_gtk_color_to_hex(gtk_color)
+  return "#" ..
+    convert_gtk_channel_to_hex(gtk_color.red) ..
+    convert_gtk_channel_to_hex(gtk_color.green) ..
+    convert_gtk_channel_to_hex(gtk_color.blue) ..
+    convert_gtk_channel_to_hex(gtk_color.alpha)
+end
+
+local function lookup_gtk_color_to_hex(_style_context, color_name)
+  local gtk_color = _style_context:lookup_color(color_name)
+  if not gtk_color then
+    return nil
+  end
+  return convert_gtk_color_to_hex(gtk_color)
+end
+
+local function get_gtk_color_property_to_hex(_style_context, property_name)
+  local state = _style_context:get_state()
+  local property = _style_context:get_property(property_name, state)
+  if not property then
+    return nil
+  end
+  return convert_gtk_color_to_hex(property.value)
+end
+
+local function read_gtk_color_properties_from_widget(gtk_widget, properties)
+  local _style_context = gtk_widget:get_style_context()
+  local result = {}
+  for result_key, style_context_property in pairs(properties) do
+    result[result_key] = get_gtk_color_property_to_hex(
+      _style_context, style_context_property
+    )
+  end
+  return result
+end
+
+
+--- Get GTK+3 theme variables from GtkStyleContext
+-- @treturn table subj
+function gtk.get_theme_variables()
+  if gtk.cached_theme_variables then
+    return gtk.cached_theme_variables
+  end
+
+  local result = {}
+  local _gtk_status, Gtk = pcall(function()
+    return require('lgi').Gtk
+  end)
+  if not _gtk_status or not Gtk then
+    gears_debug.print_warning(
+      "Can't load GTK+3 introspection. "..
+      "Seems like GTK+3 is not installed or `lua-lgi` was built with an incompatible GTK+3 version."
+    )
+    return nil
+  end
+  local _window_status, window = pcall(function()
+    return Gtk.Window{}
+  end)
+  if not _window_status or not window then
+    gears_debug.print_warning(
+      "Can't create GTK+3 window. "..
+      "Seems like GTK+3 theme is not set correctly or `lua-lgi` was built with an incompatible GTK+3 version."
+    )
+    return nil
+  end
+  local style_context = window:get_style_context()
```

Comment:
This should be `Gtk.StyleContext()` instead.

Reply:
window is used not only for retrieving the style context

Reply:
Please read your own code again and tell me where (don't count [this](https://github.com/awesomeWM/awesome/pull/2129/files/43b17d65d281f01b0ab4d65e5778c9268a4440e3#diff-45ad409b45603061b8a7d3bd863c9555R95) as `{b,f}g_color` are exported).